File: typescript/packages/management/src/tools/list_trigger_components.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const listTriggerComponents: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  return ctx.client.agents.listTriggerComponents(ctx.sandboxId, {
    app: typeof input.app === 'string' ? input.app : undefined,
    q: typeof input.q === 'string' ? input.q : undefined,
    limit: typeof input.limit === 'number' ? input.limit : undefined,
  });
};

File: python/copass-management/src/copass_management/tools/list_trigger_components.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def list_trigger_components(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    app = input.get("app")
    q = input.get("q")
    limit = input.get("limit")
    return await ctx.client.agents.list_trigger_components(
        ctx.sandbox_id,
        app=app if isinstance(app, str) else None,
        q=q if isinstance(q, str) else None,
        limit=int(limit) if isinstance(limit, int) else None,
    )
